packages/fast-usdc/src/utils/fees.js

```diff
@@ -57,26 +76,46 @@ export const makeFeeTools = (feeConfig, customFeeConfigs = harden({})) => {
      * @throws {Error} if requested does not exceed fees
      */
     calculateAdvanceFee(requested, destination) {
-      const chainId = parseCaipChainId(destination);
-      const { flat, variableRate } = customFeeConfigs[chainId] || feeConfig;
-      const fee = add(multiplyBy(requested, variableRate), flat);
+      const baseFee = feeTools.calculateBaseFee(requested, destination);
+      const relayFee = feeTools.calculateRelayFee(destination);
+      const fee = add(baseFee, relayFee);
       !isGTE(fee, requested) || Fail`Request must exceed fees.`;
       return fee;
     },
+    /**
+     * Calculate the net amount to advance after withholding fees.
+     *
+     * @param {Amount<'nat'>} requested
+     * @param {AccountId} destination
+     * @throws {Error} if requested does not exceed fees
+     */
+    calculateAdvance(requested, destination) {
+      const fee = feeTools.calculateAdvanceFee(requested, destination);
+      return subtract(requested, fee);
+    },
     /**
      * Calculate the split of fees between pool and contract.
      *
+     * The `ContractFee` includes base fees plus the relay fee.
+     *
      * @param {Amount<'nat'>} requested
      * @param {AccountId} destination
      * @returns {RepayAmountKWR} an {@link AmountKeywordRecord}
-     * @throws {Error} if requested does not exceed fees
      */
     calculateSplit(requested, destination) {
-      const fee = feeTools.calculateAdvanceFee(requested, destination);
-      const Principal = subtract(requested, fee);
-      const ContractFee = multiplyBy(fee, feeConfig.contractRate);
-      const PoolFee = subtract(fee, ContractFee);
-      return harden({ Principal, PoolFee, ContractFee });
+      const baseFee = feeTools.calculateBaseFee(requested, destination);
+      const relayFee = feeTools.calculateRelayFee(destination);
+      const totalFee = add(baseFee, relayFee);
+      !isGTE(totalFee, requested) || Fail`Request must exceed fees.`;
+
+      const chainId = parseCaipChainId(destination);
+      const { contractRate } = customFeeConfigs[chainId] || feeConfig;
+
+      const Principal = subtract(requested, totalFee);
+      const ContractFee = multiplyBy(baseFee, contractRate);
+      const PoolFee = subtract(baseFee, ContractFee);
```

Comment:
`ContractFee` has now become a bit of a misnomer... worth a note in the type? (did you already do that?)

Reply:
I missed this in my latest round - struggling to see where to best add this. `RepayAmountKWR` and `RepayPaymentKWR` aren't very visible.

Tangential - I considered tracking `totalRelayFees` separately in vstorage metrics, even though they ultimately go to the contract seat. I ran into issues since `poolMetricsRecorderKit` is constrained to `PoolMetricsShape`.

We could explore approaches to workaround the above if we think it's important.